### DISCUSSION TITLE
[Feature] Relations columns(properties) view as additional panels/subviews(hasMany and belongTo) in show action

### DISCUSSION THREAD

**Discussion by izica** [Laravel-Backpack/community-forum#326] [General]
I have been looking for a solution for a while, but I still can not find it.
Had to implement it at my project.

Model relations:
```php
    public function account_legal_entity()
    {
        return $this->belongsTo('App\Models\AccountLegalEntity');
    }

    public function account_sms_codes()
    {
        return $this->hasMany('App\Models\AccountSmsCode');
    }
```
Controller:
```php
    protected function setupShowOperation()
    {
        $this->crud->setValidation(AccountRequest::class);

        $this->crud->addColumn([
            'name'            => 'account_sms_codes',
            'label'           => 'Sms codes',
            'type'            => 'relation_has_many',
            // url piece for CRUD actions
            'backpack_entity' => 'accountsmscode',
            // set model to get from fillable
            'model'           => \App\Models\AccountSmsCode::class,
            //or set columns
            'columns'         => [
                'sms_code'   => 'Code',
                'expired_at' => 'Expired date',
            ],
        ]);

        $this->crud->addColumn([
            'name'            => 'account_legal_entity',
            'label'           => 'Legal entity',
            'type'            => 'relation_belongs_to',
            // url piece for CRUD actions
            'backpack_entity' => 'accountlegalentity',
            //optional, get from fillable by default
            'columns'         => [
                'name'   => 'Company name',
                'owner_firstname' => 'Owner firstname',
            ],
        ]);
    }
```
![Screenshot from 2020-05-01 19 45 22](https://user-images.githubusercontent.com/13401922/80806244-58fcfa80-8be4-11ea-9a36-6f44509c1e5f.png)

**Comment by welcome** [comment]
Hello there! Thanks for opening your first issue on this repo!

Just a heads-up: **Here at Backpack we use Github Issues only for tracking bugs**. Talk about new features is also acceptable. This helps _a lot_ in keeping our focus on improving Backpack. **If you issue is not a bug/feature, please help us out by closing the issue yourself and posting in the appropriate medium (see below)**. If you're not sure where it fits, it's ok, a community member will probably reply to help you with that.

Backpack communication channels:
 - Bug Reports, Feature Requests - Github Issues (here);
 - Quick help (_How do I do X_) - [Gitter Chatroom](https://gitter.im/BackpackForLaravel/Lobby);
 - Long questions (_I have done X and Y and it won't do Z wtf_) - [Stackoverflow](https://stackoverflow.com/questions/tagged/backpack-for-laravel), using the ```backpack-for-laravel``` tag;
 - Showing off something you've made, asking for opinion on Backpack/Laravel matters - [Reddit](https://www.reddit.com/r/BackpackForLaravel/);

Please keep in mind Backpack offers no official / paid support. Whatever help you receive here, on Gitter, Slack or Stackoverflow is thanks to our awesome _awesome_ community members, who give up some of their time to help their peers. If you want to join our community, just start pitching in. We take pride in being a welcoming bunch.

Thank you!

--
Justin Case
The Backpack Robot

**Comment by tabacitu** [comment]
That's really cool @izica . We were talking about going for something like that in 4.1.x - but in my mind it worked better as a [Widget](https://backpackforlaravel.com/docs/4.0/base-widgets).

Any particular reason why you built it as a column instead of a Widget? I may be missing something, but at first glance I think what you have there needs quite a few workarounds in the HTML (close table, output stand-alone column, begin table). Coding it as a widget would not need that, since you can push widgets in particular sections/stacks.

**Comment by izica** [comment]
@tabacitu Hello, my team was thinking about the Widget option, but we decided that this method is better for us (define it as a regular column).
We decided to offer this to the community.

**Comment by tabacitu** [comment]
Please do! A PR is very much appreciated - hell, even a copy-paste is better than starting from scratch 😆

**Comment by linear5ky** [comment]
Hi,
Thanks for this. It was something I was planning to incorporate in a new project. So if you could share the column code that would be appreciated :)

**Comment by izica** [comment]
@tabacitu need some refactoring before PR. it can take few days.
think need to rename types with prefix "panel_" to avoid future intersections with common fields
panel_relation_has_many
panel_relation_belongs_to

any ideas about field
'backpack_entity' => 'accountsmscode',
these fields are used to interact with crud actions, maybe there is a more elegant way to get this?

**Comment by tabacitu** [comment]
@izica no rush on our end. We won't be able to take a look at this until we launch 4.1

>think need to rename types with prefix "panel_" to avoid future intersections with common fields
panel_relation_has_many
panel_relation_belongs_to

That's one reason why I thought it would be easier to implement as a Widget 😄 

> any ideas about field
'backpack_entity' => 'accountsmscode',
these fields are used to interact with crud actions, maybe there is a more elegant way to get this?

I don't understand this - could you please rephrase?

**Comment by enefeka** [comment]
Hi, that looks really cool! Adding to @linear5ky comment, could you please share that code? Thanks!

**Comment by izica** [comment]
@tabacitu @linear5ky @enefeka 
https://github.com/Laravel-Backpack/CRUD/pull/2752

**Comment by enefeka** [comment]
Awesome, thanks!

**Comment by linear5ky** [comment]
Great work. Thank you for sharing!

**Comment by jcastroa87** [comment]
Hello.

Since this great feature became a plugin, we will close this discussion.

Thanks to @izica 

Cheers
